docs/Schema-Model.md

```diff
@@ -0,0 +1,161 @@
+i:pid: 1029714075307
+i:programid: 1
+i:programpi: Kulkarni
+i:publisher: Fink
+i:ra: 128.2831039126012
+i:ranr: 128.28545567346546
+i:rb: 0.774291
+i:rbversion: t17_f5_c3
+i:rcid: 15
+i:rfid: 569360317
+i:schemavsn: 3.3
+i:scorr: 11.468259749904854
+i:seeratio: 1.8730389
+i:sgmag1: -1.1868139E29
+i:sgmag2: 29.968622
+i:sgmag3: 29.968622
+i:sgscore1: 0.0
+i:sgscore2: 0.99584574
+i:sgscore3: 0.5
+i:sharpnr: -1.18682505E29
+i:sigmagap: 0.46825972
+i:sigmagapbig: 0.46825972
+i:sigmagnr: 0.216
+i:sigmapsf: 0.15521143
+i:simag1: -1.1868139E29
+i:simag2: 29.968622
+i:simag3: 29.968622
+i:sky: 0.5201477
+i:srmag1: 29.968622
+i:srmag2: 29.968622
+i:srmag3: -1.1868139E29
+i:ssdistnr: 0.0
+i:ssmagnr: 29.968622
+i:ssnamenr: 52124
+i:ssnrms: 13.6383
+i:sumrat: 1.8730389
+i:szmag1: -1.1868139E29
+i:szmag2: 29.968622
+i:szmag3: -1.1868139E29
+i:tblid: 22
+i:tooflag: 0
+i:xpos: 2398.9841
+i:ypos: 2400.76
+i:zpclrcov: -1.18682505E29
+i:zpmed: 29.968622
+d:cdsxmatch: Unknown
+d:rfscore: 0.0
+```
+
+The above row will be translated into graph data model as follows:
+
+### Vertices
+- Every alert (single row in hbase) will be modelled as a graph vertex.
+- Following alert properties will be modelled as vertex labels
+    - Supernova Random Forest classification ```(d:rfscore)```
+    - Supernova Bayesian Neural Network classification ```(d:snn)``` [coming soon]
+    - Microlensing Random Forest classification ```(d:mulens_class_i)``` [coming soon]
+    - Cross-match label ```(d:cdsxmatch)```
+    - Star/Galaxy separation ```(i:classtar)```
+    - The object ID itself if we use individual alerts ```(i:objectId)```
+
+### Edges
+- Create an edge between all the vertices that have same properties
+- So if Vertex 1 has ```(d:rfscore)``` property and Vertex 5 also has the same property, we will create an edge between them.
+
+
+### Data Insertion Algorithm (might change based on answers to the questions)
+
+- Start by creating the schema in janusgraph if it does not exist
+   This will involve creating 
+   - vertex labels (cf:columnname in hbase row)
+   - edge labels (not sure what will be the edge label exactly)
+   - property keys
+        - Vertex property ```Rowkey```
+- Read all the rows, group by the interesting columnnames (alert properties). Vertices in each group become ```cliques``` of the graph, since each of them needs to be connected to all the others in the group. Load all these as vertices of the graph and add the labels and properties.
+- Then for each of these groups, load edges, using the same ```Dataframe``` and add labels.
+- Create indexes
+
+### Questions
+1. How to decide edge directions? What will be the edge label?
+
+It seems it is better to go with directed edges since support for unidirected edges is limited (https://docs.janusgraph.org/advanced-topics/advschema/)
+
+2. Do we connect the vertices because they have the same property or the property and their values must match as well?
```

Comment:
For string-type property, we should match values.
For ML score (between 0 and 1), we should put a threshold (say naively 0.5). Alerts below 0.5 are of the same kind, and vice-versa. It could be more subtle later of course.